lib/elixir/lib/inspect.ex

```diff
@@ -350,9 +389,12 @@ defimpl Inspect, for: List do
   ## Private
 
   defp key_to_binary(key) do
-    case Inspect.Atom.inspect(key) do
```

Comment:
I believe this previous implementation was correct though? It would be slightly faster because we avoid traversing the atom multiple times?